### PR TITLE
fix: create tab on arc browser

### LINKED ISSUE #3: Support Arc browser (prevent open split window)
There is a problem with Arc browser opening in a split window. Moreover, it opens an extra blank page.

Kapture 2024-10-01 at 22 35 31

### PR DESCRIPTION
**Pull request by 1natsu172**
close #3 

Support for Arc. And the tab opening method change to `browser.tabs.create`. `browser.tabs.create` only works in the background, so it's necessary to use messages.

---

### Chrome 
https://github.com/user-attachments/assets/3740d682-d360-4fd0-8ae1-f19ff384dfbf  

### Firefox
https://github.com/user-attachments/assets/d5a69cda-794b-417f-a076-5689228dc1f7  

### Arc
https://github.com/user-attachments/assets/5f999cfc-9994-4e22-ac2c-4ed0b954610c

### CODE REVIEW

**Comment by ergofriend** [comment]
I didn't check the Arc Browser. Thank you for your pull request!

File: libs/messaging.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { defineExtensionMessaging } from "@webext-core/messaging";
+
+type SearchMessaging = {
+  searchOnTab: (data: { url: string }) => void;
+};
+export const searchMessaging = defineExtensionMessaging<SearchMessaging>();
```
Comment (ergofriend) [inline libs/messaging.ts:6]: wow, its cool that `defineExtensionMessaging` infer event types.

**Comment by ergofriend** [comment]
@1natsu172  If you sign the commit, I can approve!

**Comment by 1natsu172** [comment]
@ergofriend Thanks for review. Added signed commit. Is this as expected?

File: libs/messaging.ts (hunk 1 above)
Comment (1natsu172) [reply libs/messaging.ts:6]: Aaron (the author of wxt) is making this one too lol.

**Comment by ergofriend** [comment]
@1natsu172  Sorry for the poor explanation 🙏🏻 
All commits must be signed for merging.
So could you please sign the first commit? (force-push would be good)

**Comment by 1natsu172** [comment]
@ergofriend OK, I've rebased and re-signed now. Is this fine?

**Comment by ergofriend** [comment]
Perfect, thank you!

**Pull request merged**